WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Summary Section/Edit Order Status/OrderStatusListViewController.swift

```diff
@@ -57,32 +40,22 @@ final class OrderStatusListViewController: UIViewController {
         registerTableViewCells()
         configureNavigationBar()
         configureTableView()
-
-        configureResultsController()
-
-        preselectStatusIfPossible()
+        reloadTable()
     }
 
-    /// Setup: Results Controller
-    ///
-    private func configureResultsController() {
-        statusResultsController.startForwardingEvents(to: tableView)
-
-        do {
-            try statusResultsController.performFetch()
-        } catch {
-            CrashLogging.logError(error)
-        }
-
+    private func reloadTable(completion: (() -> Void)? = nil) {
+        viewModel.refreshStatuses() //X Is this necessary everytime? Seems it could be slow.
+        //X TODO - needs a completion?
         tableView.reloadData()
         preselectStatusIfPossible()
+        completion?()
```

Comment:
[Q] What is the reason for this `completion` block? since all the methods are synchronous I'm not sure what the use case for this!

Reply:
Not needed. Removed. We'll need a completion handler when pull-to-refresh is implemented in a follow on PR ( for #3608 #3609 )